react.component/src/app/Card/Card.tsx

```diff
@@ -1,58 +1,61 @@
-import { CardT } from '../cards';
+import { AppDispatch, store, fetchCharById } from '../../redux-folder';
+import { Modal } from '../Modal';
+import { BookType } from '../types';
 import './card.scss';
-import { ReactComponent as Heart } from '../../assets/heart.svg';
-import { Component, createRef, RefObject } from 'react';
+import { MouseEventHandler, useState } from 'react';
+import { useDispatch } from 'react-redux';
 
-interface NewCardT extends CardT {
-  cards: CardT[];
-  updateData: (cards: CardT[]) => void;
-};
+export const Card = (props: BookType) => {
+  const [clicked, setClicked] = useState(false);
 
-export class Card extends Component<NewCardT> {
-  heart: RefObject<HTMLDivElement>;
+  const { curChar } = store.getState().curState;
+  const dispatch = useDispatch<AppDispatch>();
 
-  constructor(props: NewCardT) {
-    super(props);
-    this.heart = createRef<HTMLDivElement>();
-  }
+  const handleCloseClick = () => setClicked(false);
+  const handleModalClick: MouseEventHandler<HTMLDivElement> = (e) => {
+    const target = e.target as HTMLDivElement;
+    if (target.dataset.testid === 'close') setClicked(false);
+  };
 
-  handleClick(event: React.MouseEvent) {
-    const heartDiv = this.heart.current;
-    const heartEl = (heartDiv ? heartDiv.firstElementChild : undefined) as SVGElement;
-    if (heartEl) {
-      heartEl.setAttribute('fill', 'darkred');
+  const handleCardClick: MouseEventHandler<HTMLDivElement> = async (e) => {
+    const target = e.target as HTMLDivElement;
+    const searchId = target.dataset.name === 'open' ? target.closest('section')?.id : null;
```

Comment:
why not just make a piece of state instead of relying on data that you've put into HTML?

Reply:
please refactor to not rely on any data attributes - everything should be stored as a state